src/templates/naive.js

```diff
@@ -100,11 +100,11 @@ function base() {
 
     function $setter(func, ...args) {
       if ($inBatch || $inRecalculate || $batchingStrategy) {
+        $batchPending.push({ func, args });
         if ((!$inBatch && !$inRecalculate) && $batchingStrategy) {
-          $batchingStrategy.call($res);
           $inBatch = true;
+          $batchingStrategy.call($res);
```

Comment:
`$inBatch = true` is now before `$batchingStrategy.call()` to allow synchronous `$batchingStrategy` to `$endBatch()`, thus setting `$inBatch` to `false`.